### PR TITLE
Add report handler

### PR DESCRIPTION
**Pull request by RachedBenMustapha**
Returns uuid, server version, system (cpu/mem/disk usage) stats, and config at the moment, minus authentication keys. The report endpoint is protected by an ephemeral token that changes on
restart and is only available to the S3 Server's children as an environment variable.

Needs scality/Arsenal#302

### CODE REVIEW

**Comment by RachedBenMustapha** [comment]
@ironman-machine 
SCALITY_INTEGRATION_BRANCH=ft/s3-report-token

**Comment by RachedBenMustapha** [comment]
@ironman-machine

**Comment by RachedBenMustapha** [comment]
@ironman-machine
SCALITY_INTEGRATION_BRANCH=ft/s3-report-token

File: circle.yml [hunk 1]
```diff
@@ -104,6 +105,10 @@ test:
             > $CIRCLE_ARTIFACTS/server_file_rawnode.txt
         & bash wait_for_local_port.bash 8000 40
         && npm run ft_node
+    - S3BACKEND=file S3VAULT=mem npm start
```
Comment (LaurenSpiegel) [inline circle.yml]: Is there any value to having the reporting work with the mem backend?

File: tests/functional/report/checkRoutes.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+const https = require('https');
+const fs = require('fs');
+
+const conf = require('../config.json');
+
+const transportStr = conf.transport;
+const transport = transportStr === 'http' ? http : https;
+
+function makeAgent() {
+    if (transportStr === 'https') {
```
Comment (LaurenSpiegel) [inline tests/functional/report/checkRoutes.js]: This currently would never be the case (only in the end to end). Do you think we should be testing this in just s3?

File: tests/functional/report/checkRoutes.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        headers: { 'x-scal-report-token': token },
+    };
+}
+
+function queryReport(done, resultCheck) {
+    const req = transport.request(options(), res => {
+        if (res.statusCode !== 200) {
+            return done(new Error(`non-200 status ${res.statusCode}`));
+        }
+
```
Comment (LaurenSpiegel) [inline tests/functional/report/checkRoutes.js]: what about checking res.on('error')?

File: circle.yml (hunk 1 above)
Comment (RachedBenMustapha) [inline circle.yml]: I don't think there's much value in monitoring a system that forgets data when it's restarted :)

File: tests/functional/report/checkRoutes.js (hunk 3 above)
Comment (RachedBenMustapha) [inline tests/functional/report/checkRoutes.js]: Good idea, will do

File: tests/functional/report/checkRoutes.js (hunk 2 above)
Comment (RachedBenMustapha) [inline tests/functional/report/checkRoutes.js]: You're right, I only realized this after I noticed that the end to end tests were only using scality's backends which won't benefit from this and the tests don't need to run there. I'll remove this.

**Comment by LaurenSpiegel** [comment]
@ironman-machine r+

**Pull request merged**